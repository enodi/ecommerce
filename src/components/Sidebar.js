import React from "react";

import CurrencySelector from "./CurrencySelector";
import { ReactComponent as RightArrow } from "../assets/right-arrow.svg";

const Sidebar = ({
  showSidebar,
  handleShowSidebar,
  itemInCart,
  decreaseCartItemCount,
  increaseCartItemCount,
  handleChangeCurrency,
  currency,
  removeItemFromCart,
  subtotal,
}) => (
  <div className={`cart-items ${showSidebar ? "show" : "hide"}`}>
    <div className="overlay" />
    <aside className="sidebar">
      <div className="sidebar-header">
        <div className="sidebar__top-section">
          <RightArrow
            style={{ width: 23, height: 23 }}
            onClick={handleShowSidebar}
            className="right-arrow"
          />
          <h4>your cart</h4>
        </div>
        <CurrencySelector
          handleChangeCurrency={handleChangeCurrency}
          currency={currency}
        />
      </div>
      <div className="sidebar-content">
        {itemInCart.length > 0 ? (
          itemInCart.map((item) => (
            <div key={`product-${item.id}`} className="sidebar__cart-item">
              <div>
                <p className="title">{item.title}</p>
                <button onClick={() => removeItemFromCart(item)}>X</button>
              </div>
              <div className="sidebar__cart-image-container">
                <img
                  src={item.image_url}
                  alt={item.title}
                  className="sidebar__cart-image"
                />
              </div>
              <div>
                <div className="quantity-regulator">
                  <button onClick={() => decreaseCartItemCount(item)}>-</button>
                  <span>{item.count}</span>
                  <button onClick={() => increaseCartItemCount(item)}>+</button>
                </div>
                <p className="price">{`${currency} ${
                  item.price * item.count
                }.00`}</p>
              </div>
            </div>
          ))
        ) : (
          <p className="empty-cart">There's no item in the cart</p>
        )}
      </div>
      {itemInCart.length > 0 && (
        <div className="sidebar__footer">
          <div className="sidebar__footer-price">
            <p>Subtotal</p>
            <p>{`${currency} ${subtotal}.00`}</p>
          </div>
          <div className="sidebar__footer-button">
            <button>revert to one time purchase</button>
            <button>proceed to checkout</button>
          </div>
        </div>
      )}
    </aside>
  </div>
);

export default Sidebar;
