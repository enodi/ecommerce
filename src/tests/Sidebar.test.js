import React from "react";
import { shallow } from "enzyme";

import { cart } from "./fixtures";
import Sidebar from "../components/Sidebar";

let handleShowSidebar,
  decreaseCartItemCount,
  increaseCartItemCount,
  handleChangeCurrency,
  removeItemFromCart,
  wrapper;
beforeEach(() => {
  handleShowSidebar = jest.fn();
  decreaseCartItemCount = jest.fn();
  increaseCartItemCount = jest.fn();
  handleChangeCurrency = jest.fn();
  removeItemFromCart = jest.fn();
  wrapper = shallow(
    <Sidebar
      showSidebar={true}
      handleShowSidebar={handleShowSidebar}
      itemInCart={cart}
      decreaseCartItemCount={decreaseCartItemCount}
      increaseCartItemCount={increaseCartItemCount}
      handleChangeCurrency={handleChangeCurrency}
      currency="NGN"
      removeItemFromCart={removeItemFromCart}
      subtotal={19400}
    />
  );
});

test("should render Sidebar component correctly", () => {
  expect(wrapper).toMatchSnapshot();
});

test("should find text rendered on sidebar", () => {
  expect(wrapper.find("h4").text()).toEqual("your cart");
});

test("should remove item from cart", () => {
  wrapper.find("button").first().simulate("click");
  expect(removeItemFromCart).toHaveBeenCalled();
});
