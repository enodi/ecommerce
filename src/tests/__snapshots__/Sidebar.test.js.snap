// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Sidebar component correctly 1`] = `
<div
  className="cart-items show"
>
  <div
    className="overlay"
  />
  <aside
    className="sidebar"
  >
    <div
      className="sidebar-header"
    >
      <div
        className="sidebar__top-section"
      >
        <ForwardRef(SvgRightArrow)
          className="right-arrow"
          onClick={[MockFunction]}
          style={
            Object {
              "height": 23,
              "width": 23,
            }
          }
        />
        <h4>
          your cart
        </h4>
      </div>
      <CurrencySelector
        currency="NGN"
        handleChangeCurrency={[MockFunction]}
      />
    </div>
    <div
      className="sidebar-content"
    >
      <div
        className="sidebar__cart-item"
        key="product-undefined"
      >
        <div>
          <p
            className="title"
          >
            Premium-Grade Moisturizing Balm
          </p>
          <button
            onClick={[Function]}
          >
            X
          </button>
        </div>
        <div
          className="sidebar__cart-image-container"
        >
          <img
            alt="Premium-Grade Moisturizing Balm"
            className="sidebar__cart-image"
            src="https://d1b929y2mmls08.cloudfront.net/luminskin/img/new-landing-page/moisturizing-balm.png"
          />
        </div>
        <div>
          <div
            className="quantity-regulator"
          >
            <button
              onClick={[Function]}
            >
              -
            </button>
            <span />
            <button
              onClick={[Function]}
            >
              +
            </button>
          </div>
          <p
            className="price"
          >
            NGN NaN.00
          </p>
        </div>
      </div>
      <div
        className="sidebar__cart-item"
        key="product-undefined"
      >
        <div>
          <p
            className="title"
          >
            No-Nonsense Charcoal Cleanser
          </p>
          <button
            onClick={[Function]}
          >
            X
          </button>
        </div>
        <div
          className="sidebar__cart-image-container"
        >
          <img
            alt="No-Nonsense Charcoal Cleanser"
            className="sidebar__cart-image"
            src="https://d1b929y2mmls08.cloudfront.net/luminskin/img/new-landing-page/charcoal-cleanser.png"
          />
        </div>
        <div>
          <div
            className="quantity-regulator"
          >
            <button
              onClick={[Function]}
            >
              -
            </button>
            <span />
            <button
              onClick={[Function]}
            >
              +
            </button>
          </div>
          <p
            className="price"
          >
            NGN NaN.00
          </p>
        </div>
      </div>
      <div
        className="sidebar__cart-item"
        key="product-undefined"
      >
        <div>
          <p
            className="title"
          >
            Clarifying Body Wash
          </p>
          <button
            onClick={[Function]}
          >
            X
          </button>
        </div>
        <div
          className="sidebar__cart-image-container"
        >
          <img
            alt="Clarifying Body Wash"
            className="sidebar__cart-image"
            src="https://i.shgcdn.com/b44f5ef8-6bc0-4a4a-8eef-1f7ced30503d/-/format/auto/-/preview/3000x3000/-/quality/lighter/"
          />
        </div>
        <div>
          <div
            className="quantity-regulator"
          >
            <button
              onClick={[Function]}
            >
              -
            </button>
            <span />
            <button
              onClick={[Function]}
            >
              +
            </button>
          </div>
          <p
            className="price"
          >
            NGN NaN.00
          </p>
        </div>
      </div>
    </div>
    <div
      className="sidebar__footer"
    >
      <div
        className="sidebar__footer-price"
      >
        <p>
          Subtotal
        </p>
        <p>
          NGN 19400
        </p>
      </div>
      <div
        className="sidebar__footer-button"
      >
        <button>
          revert to one time purchase
        </button>
        <button>
          proceed to checkout
        </button>
      </div>
    </div>
  </aside>
</div>
`;
